modern/src/store/devices.js

```diff
@@ -20,6 +21,14 @@ const { reducer, actions } = createSlice({
     remove(state, action) {
       delete state.items[action.payload];
     },
+    filterByName(state, action) {
+      return {
+        ...state,
+        filteredItems: action.payload
+          ? Object.values(state.items).filter((item) => item.name?.toLowerCase().includes(action.payload?.toLowerCase()))
```

Comment:
What happens to this if device list is updated? How is this updated?

Reply:
Every time the user changes the input text it uses the current device list to filter. That way, it will return the most updated filtered list possible.

However, if the user types "John" and stops for a few seconds and in that time new information about devices arrive, the list displayed will be outdated.

I don't think it worth the effort to make also this filtered list of devices updated at every new information arrived since it will be outdated generally for a few seconds, but it could be done if you think differently. Please, let me know what you think.

Note: if we change the filtering to the component state or to a context, this problem probably will disappear, since we will use the most updated `state.items` every time we filter.

Reply:
If you type something in the field and devices don't get updated anymore, it's a big issue.